### PR TITLE
read_ccd_data_in_interval now uses partitioning for faster accessing

### PR DESCRIPTION
**Pull request by OleMartinChristensen**
Improves performance for slow internet connections. See issue #138

### CODE REVIEW

File: src/mats_l1_processing/read_parquet_functions.py [hunk 1]
```diff
@@ -227,10 +270,23 @@ def read_ccd_data_in_interval(
                 & (ds.field(variable) <= filter[variable][1])
             )
 
+    st = time.time()
+    table = dataset.to_table(
+        filter=partition_filter & filterlist
+    )
+    et = time.time()
+    print('Execution time:', et - st, 'seconds' )
+    print(len(table))
```
Comment (skymandr) [inline src/mats_l1_processing/read_parquet_functions.py:279]:
```suggestion
    print(f"Got {len(table)} records {time.time() - st} seconds'")
```
or something like that, if you want to keep it, but are these print outs meant to be there, or are they for debugging? I don't like unsolicited output, so I would have liked to see this controlled by some sort of semaphore, in that case, but that's not the important bit.

File: src/mats_l1_processing/read_parquet_functions.py [hunk 2]
```diff
@@ -227,10 +270,23 @@ def read_ccd_data_in_interval(
                 & (ds.field(variable) <= filter[variable][1])
             )
 
+    st = time.time()
+    table = dataset.to_table(
+        filter=partition_filter & filterlist
+    )
```
Comment (skymandr) [inline src/mats_l1_processing/read_parquet_functions.py:276]:
```suggestion
    table = dataset.to_table(filter=partition_filter & filterlist)
```
looks neater, I think.

**Review by skymandr — CHANGES_REQUESTED** [review]
Two small suggestions.

Blocking since I'm not sure if you want the prints or not. Please answer that.

File: src/mats_l1_processing/read_parquet_functions.py (hunk 1 above)
Comment (skymandr) [reply src/mats_l1_processing/read_parquet_functions.py:279]: (Note: Same thing further down.)

Comment (OleMartinChristensen) [reply src/mats_l1_processing/read_parquet_functions.py]: Yes these timings should be removed (fixed now in latest patch)

File: src/mats_l1_processing/read_parquet_functions.py (hunk 2 above)
Comment (OleMartinChristensen) [reply src/mats_l1_processing/read_parquet_functions.py]: ok fixed

**Review by skymandr — COMMENTED** [review]
Looks good to me

**Comment by OleMartinChristensen** [comment]
if it looks ok now can you approve @skymandr ?

**Comment by OleMartinChristensen** [comment]
I cannot get this to work with l1b data..?

**Comment by skymandr** [comment]
> I cannot get this to work with l1b data..?

At a guess this is because the L1B-data in all current versions have a residual prefix in the form of the L1A-bucket from which it originates. This needs to be put in the path (or alternatively the prefix-attribute where applicable) explicitly. See the coverage script that I sent you for an example.

**Comment by skymandr** [comment]
> > > I cannot get this to work with l1b data..?
> > 
> > 
> > At a guess this is because the L1B-data in all current versions have a residual prefix in the form of the L1A-bucket from which it originates. This needs to be put in the path (or alternatively the prefix-attribute where applicable) explicitly. See the coverage script that I sent you for an example. Note that this prefix will be removed in future versions.
> 
> yes ill do a patch for this before we merge

I think that functionality might be better to have in Mats-utils than in this function to be fair.

**Comment by OleMartinChristensen** [comment]
> > > > I cannot get this to work with l1b data..?
> > > 
> > > 
> > > At a guess this is because the L1B-data in all current versions have a residual prefix in the form of the L1A-bucket from which it originates. This needs to be put in the path (or alternatively the prefix-attribute where applicable) explicitly. See the coverage script that I sent you for an example. Note that this prefix will be removed in future versions.
> > 
> > 
> > yes ill do a patch for this before we merge
> 
> I think that functionality might be better to have in Mats-utils than in this function to be fair.

yes, agreed. I will add it there, just want to test it before mergning in this ;)

**Pull request merged**